langchain_tools_demo/main.py

```diff
@@ -95,6 +91,19 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
         raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
 
 
+@app.post("/reset")
+async def reset(request: Request):
+    """Reset agent"""
+    global user_agents
+    uuid = request.session["uuid"]
+    if uuid in user_agents.keys():
+        await user_agents[uuid].client.close()
+        del user_agents[uuid]
+    else:
+        raise HTTPException(status_code=500, detail=f"Current agent not found")
```

Comment:
nit:

you should flip this and do:

if uuid not in X:
    raise Exception
    
# happy path here

Reply:
fixed